include/verilated_funcs.h

```diff
@@ -520,6 +520,7 @@ static inline void VL_ASSIGNBIT_WO(int bit, WDataOutP owp) VL_MT_SAFE {
             const uint32_t msb_data = VL_SEL_IWII((obits) + 1, (rwp).data(), lsb, (obits)-lsb); \
             *chunkp = msb_data & VL_MASK_E((obits)-lsb); \
         } \
+        _butemp.set(0, *(rwp).data() & 1); /* force update the sign */ \
```

Comment:
Is this the same way the existing SystemC internals do it?

Reply:
Partially. Implementation before [thread-safety fix](https://github.com/verilator/verilator/commit/1af046986d3aeb1daa7d0f96e9c17a46fe214d5f) used `range` function that internally called `set` method which updated the sign. Thread-safety fix uses different approach thus is not calling `set` function. Therefore the non-standard fix is needed to trigger sign change invoking manually `set` method.